Error building current image in main branch.

Thanks to @DLondonoD , for raising this issue.

If you try to build docker image using `docker compose build federation`: you will get next error:

```bash
[+] Building 33.4s (16/18)                                                                                   docker:default
 => [federation internal] load build definition from Dockerfile                                                        0.1s
 => => transferring dockerfile: 1.02kB                                                                                 0.0s
 => [federation] resolve image config for docker-image://docker.io/docker/dockerfile:1                                 1.6s
 => CACHED [federation] docker-image://docker.io/docker/dockerfile:1@sha256:b6afd42430b15f2d2a4c5a02b919e98a525b785b1  0.0s
 => [federation internal] load build definition from Dockerfile                                                        0.0s
 => [federation internal] load metadata for docker.io/library/alpine:latest                                            1.0s
 => [federation internal] load metadata for docker.io/library/golang:1.24.6-alpine                                     1.1s
 => [federation internal] load .dockerignore                                                                           0.0s
 => => transferring context: 2B                                                                                        0.0s
 => [federation builder 1/7] FROM docker.io/library/golang:1.24.6-alpine@sha256:c8c5f95d64aa79b6547f3b626eb84b16a7ce1  0.0s
 => [federation stage-1 1/4] FROM docker.io/library/alpine:latest@sha256:865b95f46d98cf867a156fe4a135ad3fe50d2056aa3f  0.0s
 => [federation internal] load build context                                                                           0.1s
 => => transferring context: 1.33MB                                                                                    0.1s
 => CACHED [federation builder 2/7] RUN apk add -U --no-cache ca-certificates                                          0.0s
 => CACHED [federation builder 3/7] WORKDIR /workspace                                                                 0.0s
 => [federation builder 4/7] COPY go.mod go.sum ./                                                                     0.2s
 => [federation builder 5/7] RUN --MOUNT=TYPE=SECRET,ID=NETRC,DST=/ROOT/.NETRC     go mod download                    19.2s
 => [federation builder 6/7] COPY . ./                                                                                 0.2s
 => ERROR [federation builder 7/7] RUN --mount=type=cache,target=/root/.cache/go-build      go build -o ./app ./cmd/a  9.7s
------
 > [federation builder 7/7] RUN --mount=type=cache,target=/root/.cache/go-build      go build -o ./app ./cmd/app/:
9.087 # github.com/neonephos-katalis/opg-ewbi-api/pkg/metastore
9.087 pkg/metastore/federation.go:50:24: z.ZoneId undefined (type string has no field or method ZoneId)
9.087 pkg/metastore/federation.go:51:24: z.Geolocation undefined (type string has no field or method Geolocation)
9.087 pkg/metastore/federation.go:52:24: z.GeographyDetails undefined (type string has no field or method GeographyDetails)
------
failed to solve: process "/bin/sh -c go build -o ./app ./cmd/app/" did not complete successfully: exit code: 1
```

, this problem is related with `pkg/metastore/federation.go`, changed in https://github.com/neonephos-katalis/opg-ewbi-api/commit/b1d5afc62f94d19fdd9c4d7e415f3ad6bc6a05d6 commit on main branch.

As a WA meanwhile the https://github.com/neonephos-katalis/opg-ewbi-api/pull/7 PR is merged, I suggest to modify the code with [federation.go.txt](https://github.com/user-attachments/files/24611158/federation.go.txt)

Failing again:
```bash
# docker compose build federation
WARN[0000] /home/ich/opg-ewbi-api/docker-compose.yaml: `version` is obsolete
[+] Building 6.9s (16/18)                                                                                                                                                                                                                    docker:default
 => [federation internal] load build definition from Dockerfile                                                                                                                                                                                        0.0s
 => => transferring dockerfile: 1.02kB                                                                                                                                                                                                                 0.0s
 => [federation] resolve image config for docker-image://docker.io/docker/dockerfile:1                                                                                                                                                                 0.5s
 => CACHED [federation] docker-image://docker.io/docker/dockerfile:1@sha256:2780b5c3bab67f1f76c781860de469442999ed1a0d7992a5efdf2cffc0e3d769                                                                                                           0.0s
 => [federation internal] load build definition from Dockerfile                                                                                                                                                                                        0.0s
 => [federation internal] load metadata for docker.io/library/alpine:latest                                                                                                                                                                            0.5s
 => [federation internal] load metadata for docker.io/library/golang:1.24.6-alpine                                                                                                                                                                     0.4s
 => [federation internal] load .dockerignore                                                                                                                                                                                                           0.0s
 => => transferring context: 2B                                                                                                                                                                                                                        0.0s
 => [federation builder 1/7] FROM docker.io/library/golang:1.24.6-alpine@sha256:c8c5f95d64aa79b6547f3b626eb84b16a7ce18a139e3e9ca19a8c078b85ba80d                                                                                                       0.0s
 => [federation stage-1 1/4] FROM docker.io/library/alpine:latest@sha256:25109184c71bdad752c8312a8623239686a9a2071e8825f20acb8f2198c3f659                                                                                                              0.0s
 => [federation internal] load build context                                                                                                                                                                                                           0.1s
 => => transferring context: 16.93kB                                                                                                                                                                                                                   0.0s
 => CACHED [federation builder 2/7] RUN apk add -U --no-cache ca-certificates                                                                                                                                                                          0.0s
 => CACHED [federation builder 3/7] WORKDIR /workspace                                                                                                                                                                                                 0.0s
 => CACHED [federation builder 4/7] COPY go.mod go.sum ./                                                                                                                                                                                              0.0s
 => CACHED [federation builder 5/7] RUN --MOUNT=TYPE=SECRET,ID=NETRC,DST=/ROOT/.NETRC     go mod download                                                                                                                                              0.0s
 => CACHED [federation builder 6/7] COPY . ./                                                                                                                                                                                                          0.0s
 => ERROR [federation builder 7/7] RUN --mount=type=cache,target=/root/.cache/go-build     go build -o ./app ./cmd/app/                                                                                                                                5.4s
------
 > [federation builder 7/7] RUN --mount=type=cache,target=/root/.cache/go-build     go build -o ./app ./cmd/app/:
5.007 # github.com/neonephos-katalis/opg-ewbi-api/pkg/metastore
5.007 pkg/metastore/federation.go:49:28: cannot use z (variable of struct type "github.com/neonephos-katalis/opg-ewbi-operator/api/v1beta1".ZoneDetails) as models.ZoneIdentifier value in assignment
------
failed to solve: process "/bin/sh -c go build -o ./app ./cmd/app/" did not complete successfully: exit code: 1
```

It is required to back previous code:
```golang
func federationFromK8sCustomResource(fed *opgv1beta1.Federation) (*Federation, error) {
	offeredZones := make([]models.ZoneDetails, len(fed.Spec.OfferedAvailabilityZones))
	for i, z := range fed.Spec.OfferedAvailabilityZones {
			offeredZones[i] = models.ZoneDetails{
			ZoneId: z.ZoneId,
			Geolocation: z.Geolocation,
			GeographyDetails: z.GeographyDetails,
		}
	}
```

Hi Alvaro,
You are using the old version.
Could you please change the image tag in dockercompose.yaml (line 10)?

KR
Ignazio


_________________________________________
[cid:22293515-3a05-4269-8d97-01f6d9897b95]

Ignazio Di Natali

Technology Innovation

CT.TE.TI

TIM S.p.A.
Via G. Reiss Romoli, 274 – 10148 TORINO
+39 3346062829



Gruppo TIM - Uso Interno - Tutti i diritti riservati.
________________________________
Da: afernandeztelefonica ***@***.***>
Inviato: mercoledì 8 aprile 2026 13:04
A: neonephos-katalis/opg-ewbi-api ***@***.***>
Cc: Di Natali Ignazio ***@***.***>; Mention ***@***.***>
Oggetto: [EXT] Re: [neonephos-katalis/opg-ewbi-api] Error building current image in main branch. (Issue #9)

[https://avatars.githubusercontent.com/u/194123030?s=20&v=4]afernandeztelefonica left a comment (neonephos-katalis/opg-ewbi-api#9)<https://github.com/neonephos-katalis/opg-ewbi-api/issues/9#issuecomment-4205778512>

Failing again:

# docker compose build federation
WARN[0000] /home/ich/opg-ewbi-api/docker-compose.yaml: `version` is obsolete
[+] Building 6.9s (16/18)                                                                                                                                                                                                                    docker:default
 => [federation internal] load build definition from Dockerfile                                                                                                                                                                                        0.0s
 => => transferring dockerfile: 1.02kB                                                                                                                                                                                                                 0.0s
 => [federation] resolve image config for docker-image://docker.io/docker/dockerfile:1                                                                                                                                                                 0.5s
 => CACHED [federation] ***@***.***:2780b5c3bab67f1f76c781860de469442999ed1a0d7992a5efdf2cffc0e3d769                                                                                                           0.0s
 => [federation internal] load build definition from Dockerfile                                                                                                                                                                                        0.0s
 => [federation internal] load metadata for docker.io/library/alpine:latest                                                                                                                                                                            0.5s
 => [federation internal] load metadata for docker.io/library/golang:1.24.6-alpine                                                                                                                                                                     0.4s
 => [federation internal] load .dockerignore                                                                                                                                                                                                           0.0s
 => => transferring context: 2B                                                                                                                                                                                                                        0.0s
 => [federation builder 1/7] FROM ***@***.***:c8c5f95d64aa79b6547f3b626eb84b16a7ce18a139e3e9ca19a8c078b85ba80d                                                                                                       0.0s
 => [federation stage-1 1/4] FROM ***@***.***:25109184c71bdad752c8312a8623239686a9a2071e8825f20acb8f2198c3f659                                                                                                              0.0s
 => [federation internal] load build context                                                                                                                                                                                                           0.1s
 => => transferring context: 16.93kB                                                                                                                                                                                                                   0.0s
 => CACHED [federation builder 2/7] RUN apk add -U --no-cache ca-certificates                                                                                                                                                                          0.0s
 => CACHED [federation builder 3/7] WORKDIR /workspace                                                                                                                                                                                                 0.0s
 => CACHED [federation builder 4/7] COPY go.mod go.sum ./                                                                                                                                                                                              0.0s
 => CACHED [federation builder 5/7] RUN --MOUNT=TYPE=SECRET,ID=NETRC,DST=/ROOT/.NETRC     go mod download                                                                                                                                              0.0s
 => CACHED [federation builder 6/7] COPY . ./                                                                                                                                                                                                          0.0s
 => ERROR [federation builder 7/7] RUN --mount=type=cache,target=/root/.cache/go-build     go build -o ./app ./cmd/app/                                                                                                                                5.4s
------
 > [federation builder 7/7] RUN --mount=type=cache,target=/root/.cache/go-build     go build -o ./app ./cmd/app/:
5.007 # github.com/neonephos-katalis/opg-ewbi-api/pkg/metastore
5.007 pkg/metastore/federation.go:49:28: cannot use z (variable of struct type "github.com/neonephos-katalis/opg-ewbi-operator/api/v1beta1".ZoneDetails) as models.ZoneIdentifier value in assignment
------
failed to solve: process "/bin/sh -c go build -o ./app ./cmd/app/" did not complete successfully: exit code: 1

It is required to back previous code:

func federationFromK8sCustomResource(fed *opgv1beta1.Federation) (*Federation, error) {
        offeredZones := make([]models.ZoneDetails, len(fed.Spec.OfferedAvailabilityZones))
        for i, z := range fed.Spec.OfferedAvailabilityZones {
                        offeredZones[i] = models.ZoneDetails{
                        ZoneId: z.ZoneId,
                        Geolocation: z.Geolocation,
                        GeographyDetails: z.GeographyDetails,
                }
        }

—
Reply to this email directly, view it on GitHub<https://github.com/neonephos-katalis/opg-ewbi-api/issues/9#issuecomment-4205778512>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/BILIETQS2BUCF3NNDBU26LT4UYW5XAVCNFSM6AAAAACRU3TRQ2VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHM2DEMBVG43TQNJRGI>.
You are receiving this because you were mentioned.Message ID: ***@***.***>

________________________________
Questo messaggio e i suoi allegati sono indirizzati esclusivamente alle persone indicate. La diffusione, copia o qualsiasi altra azione derivante dalla conoscenza di queste informazioni sono rigorosamente vietate. Qualora abbiate ricevuto questo documento per errore siete cortesemente pregati di darne immediata comunicazione al mittente e di provvedere alla sua distruzione, Grazie.


This e-mail and any attachments is confidential and may contain privileged information intended for the addressee(s) only. Dissemination, copying, printing or use by anybody else is unauthorized. If you are not the intended recipient, please delete this message and any attachments and advise the sender by return e-mail, Thanks.


Rispetta l'ambiente. Non stampare questa mail se non è necessario.

--- a/pkg/metastore/federation.go
+++ b/pkg/metastore/federation.go
@@ -46,7 +46,11 @@ func (f *Federation) updatek8sCustomResource(fed *opgv1beta1.Federation) *opgv1b
 func federationFromK8sCustomResource(fed *opgv1beta1.Federation) (*Federation, error) {
 	offeredZones := make([]models.ZoneDetails, len(fed.Spec.OfferedAvailabilityZones))
 	for i, z := range fed.Spec.OfferedAvailabilityZones {
-		offeredZones[i].ZoneId = z
+		offeredZones[i] = models.ZoneDetails{
+			ZoneId:           z.ZoneId,
+			Geolocation:      z.Geolocation,
+			GeographyDetails: z.GeographyDetails,
+		}
 	}
 
 	return &Federation{

